common-pg/src/main/scala/com/socrata/pg/soql/SoQLAnalysisSqlizer.scala

```diff
@@ -173,15 +173,44 @@ object SoQLAnalysisSqlizer extends Sqlizer[AnalysisTarget] {
           PostgresUniverseCommon.searchVector(primaryTableReps, Some(ctx))
       }
 
+      val searchNumberLitOpt =
+        try {
+          val number = BigDecimal.apply(searchLit.value)
+          Some(NumberLiteral(number, SoQLNumber.t)(searchLit.position))
+        } catch {
+          case _: NumberFormatException =>
+            None
+        }
+
+      val numericCols = searchNumberLitOpt match {
+        case None => Seq.empty[String]
+        case Some(searchNumberLit) =>
+          prevAna match {
```

Comment:
What is the purpose of this match block? I see a nearly identical version of this on line 169 as well, could it be DRYed?

Reply:
Your comment straddles two match.  The outer one chooses between search that is numeric and non numeric.  The inner one chooses between chained and not chained which has a similar block but dealing with the other part of search.